src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -275,6 +275,14 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
 
             return new AspireStore(Path.Combine(aspireDir, ".aspire"));
         });
+        // Add configuration values from UserSecrets even in PublishMode which is
+        // typically invoked via the CLI which doesn't set the correct environment
+        // value to load up UserSecrets. We do this because for local deploy, we want
+        // to be able to use UserSecrets as storage for deployment state.
+        if (AppHostAssembly is not null && ExecutionContext.IsPublishMode)
+        {
+            _innerBuilder.Configuration.AddUserSecrets(AppHostAssembly);
```

Comment:
>AddUserSecrets [](http://example.com/codeflow?start=40&length=14)

(super nit) would this make more sense to be higher above where we are modifying the Configuration for other things?

Reply:
Also - does this have any issues when UserSecrets is already added?

Reply:
> (super nit) would this make more sense to be higher above where we are modifying the Configuration for other things?

Sure.

> Also - does this have any issues when UserSecrets is already added?

I think this should be fine. Assuming the AppHostAssembly is the same this would just added it as a JSON config [based on this](https://source.dot.net/#Microsoft.Extensions.Configuration.UserSecrets/UserSecretsConfigurationExtensions.cs,186).